src/components/Accordion/Accordion.tsx

```diff
@@ -40,6 +40,10 @@ type AccordionButtonProps = {
    * Additional classnames passed in for styling
    */
   className?: string;
+  /**
+   * Icon override for component. Default is 'expand-more'
+   */
+  icon?: Extract<IconName, 'expand-more'>;
```

Comment:
This might not look all that useful but hear me out :)

Firstly, this is a first step in creating a new union utility type that replaces `expand-more` with `T` along with some other functionality. that will allow for injecting custom icons with some constraints. There is some architecture work to do for that final state so replacing the cases now in anticipation of that functionality. Using [`Extract`](https://www.typescriptlang.org/docs/handbook/utility-types.html#extracttype-union) like this adds some ergonomics for when the the available icons change in the future, or when teams provide their own Icon sets (and thus their own `IconName` potentially)

* Allow for a generic Icon component wrapper with useful constraints (control of viewport, enforce accessibility, etc.) in the type.
* Implement utility type, likely in `Icons.tsx` as a new export
* Find/replace all usages of this construction with that utility type where needed (searching "Extract<IconName," makes that easy)

Reply:
Eventually this will allow some subset of icons?

Probably easier to allow any icon name be used (at the expense of control).

Reply:
This is one of the cases where it'd either be this specific built-in icon, or some custom icon that team wants to use (and not another EDS icon) when I work on the second part.

In the next week or two, hoping to have a chat with some design folks so I can start on the second part of this one, and possibly simplify parts of this further